feature/experimental/system/gnmi/benchmarking/otg_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_isis_test.go

```diff
@@ -0,0 +1,141 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// drained_configuration_convergence_time_test is used to verify isis test scenarios
+// as given in gnmi 1.3 testcase
+package drained_configuration_convergence_time_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/feature/experimental/system/gnmi/benchmarking/otg_tests/internal/setup"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	otgtelemetry "github.com/openconfig/ondatra/gnmi/otg"
+	"github.com/openconfig/ygnmi/ygnmi"
+)
+
+// setISISOverloadBit is used to configure isis overload bit to true.
+func setISISOverloadBit(t *testing.T, dut *ondatra.DUTDevice) {
+
+	// ISIS Configs to set Overload Bit to true.
+	dutISISPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, setup.ISISInstance).Isis()
+	lspBit := dutISISPath.Global().LspBit().OverloadBit()
+	gnmi.Replace(t, dut, lspBit.SetBit().Config(), true)
+}
+
+// setISISMetric is used to configure metric on isis interfaces.
+func setISISMetric(t *testing.T, dut *ondatra.DUTDevice) {
+
+	dutISISPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, setup.ISISInstance).Isis()
+	t.Logf("Configure ISIS metric to %v", setup.ISISMetric)
+	for _, dp := range dut.Ports() {
+		intfName := dp.Name()
+		if *deviations.ExplicitInterfaceInDefaultVRF {
+			intfName = dp.Name() + ".0"
+		}
+		dutISISPathIntfAF := dutISISPath.Interface(intfName).Level(2).Af(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST)
+		gnmi.Replace(t, dut, dutISISPathIntfAF.Metric().Config(), setup.ISISMetric)
+	}
+}
+
+// verifyISISMetric is used to verify on ATE to see how much time it
+// has taken to apply changes in metric
+func verifyISISMetric(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice) {
+
+	t.Run("ISIS Metric verification", func(t *testing.T) {
+		for _, ap := range ate.Ports() {
+			if ap.ID() == "port1" {
+				// Port1 is ingress, skip verification on ingress port
+				continue
+			}
+
+			_, ok := gnmi.WatchAll(t, ate.OTG(), gnmi.OTG().IsisRouter("devIsis"+ap.Name()).LinkStateDatabase().LspsAny().Tlvs().ExtendedIpv4Reachability().PrefixAny().Metric().State(), time.Minute, func(v *ygnmi.Value[uint32]) bool {
+				metric, present := v.Val()
+				if present {
+					if metric == setup.ISISMetric {
+						return true
+					}
+				}
+				return false
+			}).Await(t)
+
+			metricInReceivedLsp := gnmi.GetAll(t, ate.OTG(), gnmi.OTG().IsisRouter("devIsis"+ap.Name()).LinkStateDatabase().LspsAny().Tlvs().ExtendedIpv4Reachability().PrefixAny().Metric().State())[0]
```

Comment:
Can just get the last value from the WatchAll calls first returned value, shouldn't need to use the additional GetAll here.

Reply:
yeah agree, changed